src/glmfit.jl

```diff
@@ -507,13 +507,20 @@ function dispersion(m::AbstractGLM, sqr::Bool=false)
 end
 
 """
-    predict(mm::AbstractGLM, newX::AbstractMatrix; offset::FPVector=Vector{eltype(newX)}(0))
-
-Form the predicted response of model `mm` from covariate values `newX` and, optionally,
-an offset.
+    predict(mm::AbstractGLM, newX::AbstractMatrix; offset::FPVector=eltype(newX)[],
+            interval::Union{Symbol,Nothing}=nothing, level::Real = 0.95,
+            symmetric::Bool = false)
+
+If `interval` is `nothing` (the default), return the predicted response of model
+`mm` from covariate values `newX` and, optionally, an `offset`. If `interval` is
+`:confidence`, also return upper and lower bounds for a given coverage `level`. By 
+default (`symmetric = false`) the intervals are constructed by applying the inverse link
+to intervals for the linear predictor, if `symmetric=true`, the intervals are symmetric 
```

Comment:
That is true. I guess it depends on what you mean by "correct" :D. These intervals have correct coverage under old school asymptotics (number of features fixed, number of samples to infinity, design matrix well-behaved...). I know that does not mean a lot, but I would not rigorously trust the "asymmetric" intervals either. 

I am hoping we can keep this option as my original goal for this PR was to reproduce a figure from [CASI](https://web.stanford.edu/~hastie/CASI/) which had symmetric intervals around the logistic predictions. If you don't like this, maybe we can instead also provide a standard error option like R does? Although that would make the interface clunkier I think.

Reply:
OK, if Efron and Hastie use that... :-D

Does this approach have a name in the literature or existing software though? "symmetric" is rather the consequence of computing the interval on the response scale instead of the transformed scale, but it doesn't really say what's the difference (why would somebody prefer non-symmetric intervals? :-p).

Reply:
Ok so after rereading that part, Efron/Hastie plot predictions +- 1 standard error and say it's the standard error (they don't call it a confidence interval or anything like that).
  
I searched a bit more:
* It appears that STATA (https://www.stata.com/support/faqs/statistics/standard-error-predicted-probability/) uses "symmetric" intervals (though they don't give a name to these).
*  Cox, C., & Ma, G. (1995). "Asymptotic confidence bands for generalized nonlinear regression models". Biometrics, 142-150. use the name " asymmetric intervals" for the transformation intervals (I agree that's not good though).
* [Xu and Long (2005)](https://journals.sagepub.com/doi/pdf/10.1177/1536867X0500500405) use the names "endpoint transformation intervals" and "delta method intervals" for the asymmetric, resp. symmetric intervals. I think that sounds more reasonable?

One proposal then would be to change the keyword argument from `symmetric` to `interval_method` and then in the docs:

#
By default (`interval_method = :endpoint_transformation`) the intervals are constructed by applying the inverse link
to intervals for the linear predictor. If `interval_method = :delta`, the intervals are constructed by the delta method, i.e., by linearization of the predicted response around the linear predictor. The `:delta` method intervals are symmetric around the point estimates, but do not respect natural parameter constraints (e.g., the lower bound for a probability could be negative).
# 
What do you think?

Reply:
OK, `:delta` indeed sounds like a better description. Though you mention that method in a comment that applies to both branches in the code so it probably needs adjusting.

For the default method, maybe just `:endpoint_transformation` is quite long? Maybe `:transformation` would be enough? Or maybe we can find a term that describes how the values are computed before the transformation?

Reply:
Sounds good! I will make sure to adjust that comment when I update the file.  `:transformation` sounds good to me as well. (I don't have a good idea about how to name it any other way; before transformation it's just x'\hat{\beta}, so just a linear transformation of the "approximately" Gaussian \hat{\beta}.)